lib/xml_builder.ex

```diff
@@ -214,6 +214,9 @@ defmodule XmlBuilder do
   def generate(any, options \\ []),
     do: format(any, 0, options) |> IO.chardata_to_string
 
+  defp format(:xml_decl, 0, [encoding: encoding]),
```

Comment:
If it pattern matches on the exact array `[encoding: encoding]`, it won't work if `[encoding: ..., format: ..]` is passed. Maybe just keep one `fomat(:xml_decl` and pull out the option instead of pattern match.

```elixir
  defp format(:xml_decl, 0, opts) do
      encoding = Keyword.get(opts, :encoding, "UTF-8")
      
      ~s|<?xml version="1.0" encoding="#{encoding}"?>|
  end
```